packages/dx-react-grid/src/plugins/column-chooser.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { findDOMNode } from 'react-dom';
+import {
+  Template, TemplatePlaceholder, PluginContainer, TemplateConnector,
+} from '@devexpress/dx-react-core';
+
+import { getMessagesFormatter, columnChooserItems } from '@devexpress/dx-grid-core';
+
+const pluginDependencies = [
+  { pluginName: 'TableColumnVisibility' },
+  { pluginName: 'Toolbar' },
+];
+export class ColumnChooser extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: false,
+    };
+
+    this.handleClickButton = this.handleClickButton.bind(this);
+    this.handleHide = this.handleHide.bind(this);
+  }
+  handleClickButton() {
+    this.setState({
+      visible: true,
+      overlayTarget: findDOMNode(this.button), // eslint-disable-line react/no-find-dom-node
+    });
+  }
+  handleHide() {
+    this.setState({
+      visible: false,
+    });
+  }
+  render() {
+    const {
+      overlayComponent: Overlay,
+      containerComponent: Container,
+      itemComponent: Item,
+      toggleButtonComponent: ToggleButton,
+      messages,
+    } = this.props;
+    const getMessage = getMessagesFormatter(messages);
+    const { visible, overlayTarget } = this.state;
+    return (
+      <PluginContainer
+        pluginName="ColumnChooser"
+        dependencies={pluginDependencies}
+      >
+        <Template name="toolbarContent">
+          <TemplatePlaceholder />
+          <TemplateConnector>
+            {({ columns, hiddenColumns }, { toggleVisibility }) => ([
+              <ToggleButton
+                key="toggleButton"
+                ref={(button) => { this.button = button; }}
```

Comment:
Avoid using refs in plugin

Reply:
@kvet If I move `ref` to the `ToggleButton` component. Is this way better?

Reply:
Yes. We may use refs in plugin components. But the plugin should remain "pure"